generator/src/main/java/org/apache/kafka/message/SchemaGenerator.java

```diff
@@ -270,22 +272,22 @@ private String fieldTypeToSchemaType(FieldType type,
             if (fieldFlexibleVersions.contains(version)) {
                 return nullable ? "Type.COMPACT_NULLABLE_BYTES" : "Type.COMPACT_BYTES";
             } else {
-                return nullable ? "Type.NULLABLE_BYTES" : "Type.BYTES";
+                return zeroCopy ? "Type.BYTE_BUFFER" : (nullable ? "Type.NULLABLE_BYTES" : "Type.BYTES");
```

Comment:
Like I commented earlier, whether zero copy is enabled does not affect the type.  Please get rid of Type.BYTE_BUFFER, since it is redundant.

Reply:
fixed.